BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -0,0 +1,594 @@
+
+        ParamDesc<Point2i> ColorResolutionDesc
+        {
+            get
+            {
+                ParamDesc<Point2i> res = new ParamDesc<Point2i>();
+                res.Unit = "Pixel";
+                res.Description = "Resolution of the color sensor.";
+                res.ReadableWhen = ParamDesc.ConnectionStates.Connected | ParamDesc.ConnectionStates.Disconnected;
+                res.WritableWhen = ParamDesc.ConnectionStates.Disconnected;
+                return res;
+            }
+        }
+
+        private uint _colorFPS = 30;
+        public uint ColorFPS
+        {
+            get => _colorFPS;
+            set => _colorFPS = value;
+        }
+
+        ParamDesc<Point2i> ColorFPSDesc
+        {
+            get
+            {
+                ParamDesc<Point2i> res = new ParamDesc<Point2i>();
+                res.Unit = "Frames per Second";
+                res.Description = "FPS of the color sensor.";
+                res.ReadableWhen = ParamDesc.ConnectionStates.Connected | ParamDesc.ConnectionStates.Disconnected;
+                res.WritableWhen = ParamDesc.ConnectionStates.Disconnected;
+                return res;
+            }
+        }
+
+        Point2i _depthResolution = new Point2i(640, 480);
+        public Point2i DepthResolution
+        {
+            get => _depthResolution;
+            set => _depthResolution = value;
+        }
+
+        ParamDesc<Point2i> DepthResolutionDesc
+        {
+            get
+            {
+                ParamDesc<Point2i> res = new ParamDesc<Point2i>();
+                res.Unit = "Pixel";
+                res.Description = "Resolution of the depth sensor.";
+                res.ReadableWhen = ParamDesc.ConnectionStates.Connected | ParamDesc.ConnectionStates.Disconnected;
+                res.WritableWhen = ParamDesc.ConnectionStates.Disconnected;
+                return res;
+            }
+        }
+
+        private uint _depthFPS = 30;
+        public uint DepthFPS
+        {
+            get => _depthFPS;
+            set => _depthFPS = value;
+        }
+
+        ParamDesc<Point2i> DepthFPSDesc
+        {
+            get
+            {
+                ParamDesc<Point2i> res = new ParamDesc<Point2i>();
+                res.Unit = "Frames per Second";
+                res.Description = "FPS of the depth sensor.";
+                res.ReadableWhen = ParamDesc.ConnectionStates.Connected | ParamDesc.ConnectionStates.Disconnected;
+                res.WritableWhen = ParamDesc.ConnectionStates.Disconnected;
+                return res;
+            }
+        }
+
+        #region Constructor
+        public RealSense2() : base()
+        {
+            _context = RealSense2API.CreateContext();
+            _pipeline = RealSense2API.CreatePipeline(_context);
+            _config = RealSense2API.CreateConfig();
+
+            RealSense2API.DisableAllStreams(_config);
+        }
+
+        ~RealSense2()
+        {
+            RealSense2API.DeleteConfig(_config);
```

Comment:
@f00f It depends.

If `DeleteConfig` frees or releases unmanaged stuff (and *only* unmanaged stuff), then this is partially correct. `IDisposable::Dispose()` should still be implemented to call this in a deterministic fashion, but it is at least not wrong. If there's any managed stuff released by `DeleteConfig()`, this is wrong and could potentially lead to odd crashes because garbage collection order is never guaranteed by .NET. That managed stuff could be gone before `~RealSense2()` gets called.

Either way, it is mandatory to implement the `Dispose` pattern properly. And this isn't it.

Reply:
fixed